Signal/src/Models/OWSContactAvatarBuilder.m

```diff
@@ -48,7 +49,14 @@ - (instancetype)initWithSignalId:(NSString *)signalId
                         diameter:(NSUInteger)diameter
                  contactsManager:(OWSContactsManager *)contactsManager
 {
-    NSString *name = [contactsManager displayNameForPhoneIdentifier:signalId];
+    // Name for avatar initialis.
+    NSString *_Nullable name = [contactsManager nameFromSystemContactsForRecipientId:signalId];
+    if (name.length == 0) {
+        name = [contactsManager profileNameForRecipientId:signalId];
+    }
+    if (name.length == 0) {
+        name = signalId;
```

Comment:
This will generate an avatar of "#"


> IMO we should change that to be a user icon.

Like a generic user icon you mean? We talked about this once months ago, but opted for consistency with Signal-Desktop/Android.